devicetypes/smartthings/zwave-siren.src/zwave-siren.groovy

```diff
@@ -129,25 +136,25 @@ def poll() {
 	}
 }
 
-private Boolean secondsPast(timestamp, seconds) {
-	if (!(timestamp instanceof Number)) {
-		if (timestamp instanceof Date) {
-			timestamp = timestamp.time
-		} else if ((timestamp instanceof String) && timestamp.isNumber()) {
-			timestamp = timestamp.toLong()
-		} else {
-			return true
-		}
-	}
-	return (new Date().time - timestamp) > (seconds * 1000)
-}
-
 def on() {
 	log.debug "sending on"
-	[
-		zwave.basicV1.basicSet(value: 0xFF).format(),
-		zwave.basicV1.basicGet().format()
-	]
+	// ICP-5323: Zipato siren sometimes fails to make sound for full duration
+	// Those alarms do not end with Siren Notification Report.
+	// For those cases we add additional state check after alarm duration to
+	// synchronize cloud state with actual device state.
+	if (zwaveInfo.mfr == "0258" && zwaveInfo.model == "1088") {
+		[
+			zwave.basicV1.basicSet(value: 0xFF).format(),
+			zwave.basicV1.basicGet().format(),
```

Comment:
Why is this doing double get? why not just do 2nd get?

Reply:
Actually, wouldn't a runIn be more appropriate for the later state check?

Reply:
My guess is that the device doesn't send the report in response to the set. The first get is to get the "on", the second get is to get the "off".